ietf/doc/templatetags/ietf_filters.py

```diff
@@ -613,3 +613,80 @@ def action_holder_badge(action_holder):
     else:
         return ''  # no alert needed
 
+@register.filter
+def is_regular_agenda_item(assignment):
+    """Is this agenda item a regular session item?
+    
+    A regular item appears as a sub-entry in a timeslot within the agenda
+
+    >>> from collections import namedtuple  # use to build mock objects
+    >>> factory = lambda t: namedtuple('t1', ['timeslot'])(timeslot=namedtuple('t2', ['type'])(type=namedtuple('t3', ['slug'])(slug=t)))
+    >>> is_regular_agenda_item(factory('regular'))
+    True
+
+    >>> any(is_regular_agenda_item(factory(t)) for t in ['plenary', 'break', 'reg', 'other', 'officehours'])
+    False
+    """
+    return assignment.timeslot.type.slug == 'regular'
```

Comment:
Good question.

This strikes me as a bit of a corner case for Demeter because the explicit purpose of the awkwardly named `SchedTimeSessAssignment` model is to link a session with a timeslot. Though not explicitly a method, `timeslot` is part of the public interface of the class. Python allows us to replace it with a `@property` getter method returning a `TimeSlot` if we need to change the inner workings.

On the other hand, I find it a bit arbitrary that the `TimeSlot` and not the `Session` defines the purpose of the assignment. It's not hard to imagine the rules for an assignment becoming more complicated. That, I think, argues for a method on `SchedTimeSessAssignment` being responsible for determining the type.

Naming, as usual, becomes a problem. Calling it `slot_type` is logical but the name suggests that it's just a pass-through to the timeslot. Calling it `type` would get in the way if there are ever different types of assignments (like "tentative" vs "fixed", e.g.). I'm not sure how to describe "the type of event represented by this assignment" succinctly. On the other other hand, it's returning a `TimeSlotTypeName`, so the relationship to a slot type is pretty well baked in.

Ultimately I agree it should be a method on the assignment, and I think calling it `slot_type()` is as sensible as anything.

Similarly, I will add a `meeting()` method to the class. I'm going to let references to `assignment.session.name` stand because I think the `Session` should be in charge of its name.